core/capability-negotiation.md

```diff
@@ -0,0 +1,437 @@
+is received.
+
+Example:
+
+    Client: CAP REQ :multi-prefix sasl
+    Server: CAP * ACK :multi-prefix sasl
+
+### The CAP ACK subcommand
+
+The ACK subcommand is sent by the server to acknowledge a client-sent REQ, and let the client
+know that their requested capabilities have been enabled.
+
+This subcommand has a single parameter consisting of a space-separated list of capability
+names. Each capability name may be prefixed with a dash (`-`), indicating that this
+capability has been disabled as requested.
+
+### The CAP NAK subcommand
+
+The NAK subcommand designates that the requested capability change was rejected.  The server
+MUST NOT make any change to any capabilities if it replies with a NAK subcommand.
+
+The argument of the NAK subcommand MUST consist of at least the first 100 characters of the
+capability list in the REQ subcommand which triggered the NAK.
+
+Example:
+
+    Client: CAP REQ :multi-prefix sasl ex3
+    Server: CAP * NAK :multi-prefix sasl ex3
+
+### The CAP END subcommand
+
+The END subcommand signals to the server that capability negotiation is complete and requests
+that the server continue with client registration. If the client is already registered, this
+command MUST be ignored by the server.
+
+Clients that support capabilities but do not wish to enter negotiation SHOULD send CAP END
+upon connection to the server.
+
+### The CAP NEW subcommand
+
+The NEW subcommand MUST ONLY be sent to clients that have negotiated CAP Version `302` or
+enabled the `cap-notify` capability.
+
+The NEW subcommand signals that the server supports one or more new capabilities, and may be
+sent at any time. Clients that support `CAP NEW` messages should respond with a `CAP REQ`
+message if they wish to enable one of the newly-offered capabilities.
+
+The format of a `CAP NEW` message is:
+
+    CAP <nick> NEW :<extension 1> [<extension 2> ... [<extension n>]]
+
+As with `LS`, the last parameter is a space-separated list of new capabilities that are now
+offered by the server. If the client supports CAP Version `302`, the capabilities SHOULD be
+listed with values, as they would be if the client sent `CAP LS 302`.
+
+Example:
+
+    Server: :irc.example.com CAP modernclient NEW :batch
+
+Example with following `REQ`:
+
+    Server: :irc.example.com CAP tester NEW :away-notify extended-join
+    Client: CAP REQ :extended-join away-notify
+    Server: :irc.example.com CAP tester ACK :extended-join away-notify
+
+### The CAP DEL subcommand
+
+The NEW subcommand MUST ONLY be sent to clients that have negotiated CAP Version `302` or
+enabled the `cap-notify` capability.
+
+The DEL subcommand signals that the server no longer supports one or more capabilities that
+have been advertised. Upon receiving a `CAP DEL` message, the client MUST treat the listed
+capabilities cancelled and no longer available. Clients SHOULD NOT send `CAP REQ` messages
+to cancel the capabilities in `CAP DEL`, as they have already been canceled by the server.
```

Comment:
> cancelled

Brit vs American spelling. We don't have a style guide for this, but should be consistent within one document at least.